### PR TITLE
Added logger statements in get_schema function

### PR DESCRIPTION
**Pull request by vijaypavann-db**
Added logger statements in Metadata (Schema) fetch functions.
* logging `Schema Query`, `start time` and `end time`.

### CODE REVIEW

File: src/databricks/labs/remorph/reconcile/connectors/snowflake.py [hunk 1]
```diff
@@ -79,7 +82,10 @@ def get_schema(
             SnowflakeDataSource._SCHEMA_QUERY.format(catalog=catalog, schema=schema, table=table),
         )
         try:
+            logger.debug(f"Fetching schema using query: \n`{schema_query}`")
+            logger.warning(f"Fetching Schema: Started at: {datetime.now()}")
```
Comment (sundarshankar89) [inline src/databricks/labs/remorph/reconcile/connectors/snowflake.py:86]:
```suggestion
            logger.info(f"Fetching Schema: Started at: {datetime.now()}")
```
change it to info for both.

**Review by sundarshankar89 — CHANGES_REQUESTED** [review]

File: src/databricks/labs/remorph/reconcile/connectors/snowflake.py [hunk 2]
```diff
@@ -67,6 +69,7 @@ def read_data(
         except (RuntimeError, PySparkException) as e:
             return self.log_and_throw_exception(e, "data", table_query)
 
+    @timeit
```
Comment (sundarshankar89) [inline src/databricks/labs/remorph/reconcile/connectors/snowflake.py:72]: Don't add `timeit` here

**Review by sundarshankar89 — CHANGES_REQUESTED** [review]
Small nit:

File: src/databricks/labs/remorph/reconcile/connectors/snowflake.py (hunk 2 above)
Comment (vijaypavann-db) [reply src/databricks/labs/remorph/reconcile/connectors/snowflake.py]: Done

**Comment by vijaypavann-db** [comment]
Snowflake documentation doesn't describe any user-configurable partitioning or clustering for Information Schema views. These views appear to be system-managed and optimized by Snowflake internally.

* Partitioning
  https://docs.snowflake.com/en/sql-reference/info-schema#considerations-for-replacing-show-commands-with-information-schema-views
  "The Information Schema views are optimized for queries that retrieve a small subset of objects from the dictionary. Whenever possible, maximize the performance of your queries by filtering on schema and object names."

* Data 
  https://docs.snowflake.com/en/sql-reference/identifiers-syntax#double-quoted-identifiers
  
  - Only unquoted identifiers are treated as case-insensitive and are stored in uppercase in INFORMATION_SCHEMA
  - Quoted identifiers are stored as it is:
   
    image

    **In our codebase, we're trying to convert the Schema to uppercase which may not work for the cases like above.**
    `SnowflakeDataSource._SCHEMA_QUERY.format(catalog=catalog, schema=schema.upper(), table=table)`

   - Schema converted to Upper Case: 
     image

   - The same is applicable for Table Identifier as well: 
     image

  - Lower Table doesn't work: 
     image

**Review by sundarshankar89 — DISMISSED** [review]
LGTM

File: src/databricks/labs/remorph/config.py [hunk 3]
```diff
@@ -85,6 +83,25 @@ class DatabaseConfig:
     target_schema: str
     source_catalog: str | None = None
 
+    def __post_init__(self):
+        # Source systems: Oracle, Snowflake store metadata in upper case
+        self.source_schema = (
```
[the remark this answers is not in the document]
Comment (sundarshankar89) [reply src/databricks/labs/remorph/config.py:88]: We need this revisited.

**Review by sundarshankar89 — CHANGES_REQUESTED** [review]

File: src/databricks/labs/remorph/config.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (vijaypavann-db) [reply src/databricks/labs/remorph/config.py:88]: If the user-provided input contains any special characters, we could escape them with `backticks`, but what if the catalog is created with quoted identifiers, e.g., "lowercase_catalog," and it's stored as it is (lowercase) in the INFORMATION_SCHEMA?

If we handle the user input with escape characters in the codebase, we would miss these kinds of cases.

File: src/databricks/labs/remorph/reconcile/connectors/snowflake.py [hunk 4]
```diff
@@ -17,13 +18,30 @@
...
+          - see https://docs.snowflake.com/en/sql-reference/info-schema#considerations-for-replacing-show-commands-with-information-schema-views
+       * DATA:
+          - only unquoted identifiers are treated as case-insensitive and are stored in uppercase.
+          - for quoted identifiers refer:
+             https://docs.snowflake.com/en/sql-reference/identifiers-syntax#double-quoted-identifiers
+       * ORDINAL_POSITION:
+          - indicates the sequential order of a column within a table or view, 
+             starting from 1 based on the order of column definition.
+    """
+    _SCHEMA_QUERY = """select column_name,
```
[the remark this answers is not in the document]
Comment (vijaypavann-db) [reply src/databricks/labs/remorph/reconcile/connectors/snowflake.py:32]: Done

**Review by sundarshankar89 — APPROVED** [review]
LGTM

**Pull request merged**